### PR TITLE
DEV-5140 Added documentation to README.rst

### PR DESCRIPTION
**Pull request by nathan-eiq**
Part of [DEV-5140](https://eclecticiq.atlassian.net/browse/DEV-5140)

    Added documentation to README.rst

### CODE REVIEW

File: extensions/greynoise/incoming_feeds/greynoise_noise_incoming_feed_v2/README.rst [hunk 1]
```diff
@@ -1,39 +1,639 @@
...
 Configuration options
+=====================
+
+..  csv-table::
+    :align: left
+    :header-rows: 1
+    :stub-columns: 1
+
+    "title","name","type","required","description","default"
+    "API URL","api_url","string","true","The URL pointing to the API endpoint exposing the service that makes the data available for retrieval through the feed.","https://api.greynoise.io/v2"
```
Comment (petarc-eiq) [inline extensions/greynoise/incoming_feeds/greynoise_noise_incoming_feed_v2/README.rst:56]: default="https://api.greynoise.io/"

File: extensions/greynoise/incoming_feeds/greynoise_noise_incoming_feed_v2/README.rst [hunk 2]
```diff
@@ -1,39 +1,639 @@
...
+    "API Key","api_key","string","true","API Key","***"
+    "Classification","classification","array","true","Include indicators with selected classification",""
```
Comment (petarc-eiq) [inline extensions/greynoise/incoming_feeds/greynoise_noise_incoming_feed_v2/README.rst:58]: default="malicious"

File: extensions/greynoise/incoming_feeds/greynoise_noise_incoming_feed_v2/README.rst [hunk 3]
```diff
@@ -1,39 +1,639 @@
...
+
+
+Supported Observable types
```
Comment (petarc-eiq) [inline extensions/greynoise/incoming_feeds/greynoise_noise_incoming_feed_v2/README.rst:61]: @nathan-eiq  Supported Observable types means "which observable types can this enricher process?", so it doesn't fit here. I don't think it's meant to be "which observable types does this extension provide in results?". @zed-eiq do you agree?

**Review by petarc-eiq — CHANGES_REQUESTED** [review]

File: extensions/greynoise/incoming_feeds/greynoise_noise_incoming_feed_v2/README.rst (hunk 3 above)
Comment (nathan-eiq) [reply extensions/greynoise/incoming_feeds/greynoise_noise_incoming_feed_v2/README.rst:61]: That is the idea behind this section: "which observable types are provided by this extension?"

Comment (zed-eiq) [reply extensions/greynoise/incoming_feeds/greynoise_noise_incoming_feed_v2/README.rst:61]: ah, no, petar is right. this section is usually present in enricher docs, because enrichers have specific observable types defined in code that it can enrich.

incoming feeds/outgoing feeds shouldn't have this section -- info that you're thinking of expressing here should be covered in the rest of the doc already

tl;dr: remove :grin:

Comment (nathan-eiq) [reply extensions/greynoise/incoming_feeds/greynoise_noise_incoming_feed_v2/README.rst]: Done

File: extensions/greynoise/incoming_feeds/greynoise_noise_incoming_feed_v2/README.rst [hunk 4]
```diff
@@ -1,39 +1,624 @@
...
+    "Classification","classification","array","true","Include indicators with selected classification","malicious"
+
+Data Mapping
+============
+
+Entities are transformed from GNQL json to `STIX 2.1 <https://docs.oasis-open.org/cti/stix/v2.1/csprd01/stix-v2.1-csprd01.html>`_ entities.
+
+GreyNoise Identity SDO
+^^^^^^^^^^^^^^^^^^^^^^
+
```
Comment (zed-eiq) [inline extensions/greynoise/incoming_feeds/greynoise_noise_incoming_feed_v2/README.rst:66]: Add

> All STIX 2.1 objects produced by this extension will reference this Identity SDO as its producer  (e.g. in their `created_by_ref` field).

File: extensions/greynoise/incoming_feeds/greynoise_noise_incoming_feed_v2/README.rst [hunk 5]
```diff
@@ -1,39 +1,624 @@
...
+    * - name
+      - ``N/A``
+      - ``GreyNoise``
+
+    * - identity_class
+      - ``N/A``
+      - ``organization``
+
+Ipv4 SCO
```
Comment (zed-eiq) [inline extensions/greynoise/incoming_feeds/greynoise_noise_incoming_feed_v2/README.rst:104]: Group all SCO objects together under an SCO heading

File: extensions/greynoise/incoming_feeds/greynoise_noise_incoming_feed_v2/README.rst [hunk 6]
```diff
@@ -1,39 +1,624 @@
...
+      - ``item["metadata"].get("organization")``
+
+    * - identity_class
+      - ``N/A``
+      - ``organization``
+
+.. Note::
+  Only one identity can be created for each ``organization`` value. Reuse or update the object when it is seen subsequently.
+
+IPv4 SCO to Indicator SDO - SRO
```
Comment (zed-eiq) [inline extensions/greynoise/incoming_feeds/greynoise_noise_incoming_feed_v2/README.rst:425]: group all SRO objects together under a common SRO heading

File: extensions/greynoise/incoming_feeds/greynoise_noise_incoming_feed_v2/README.rst [hunk 7]
```diff
@@ -1,39 +1,624 @@
...
+
+..  list-table::
+    :header-rows: 1
+    :stub-columns: 1
+
+    * - Field Name
+      - GNQL JSON
+      - STIX 2.1
+
+    * tags
```
Comment (zed-eiq) [inline extensions/greynoise/incoming_feeds/greynoise_noise_incoming_feed_v2/README.rst:203]: broken table

File: extensions/greynoise/incoming_feeds/greynoise_noise_incoming_feed_v2/README.rst [hunk 8]
```diff
@@ -1,39 +1,624 @@
...
+      - ``data.tags``
+      - ``item.get("tags")
```
Comment (zed-eiq) [inline extensions/greynoise/incoming_feeds/greynoise_noise_incoming_feed_v2/README.rst:205]: is this correct? if they're all going to the `labels` field of the indicator SDO object, then there shouldn't be a STIX 2.1 field named here (also, these don't look like STIX 2.1 fields)

File: extensions/greynoise/incoming_feeds/greynoise_noise_incoming_feed_v2/README.rst [hunk 9]
```diff
@@ -1,39 +1,624 @@
...
+      - ``N/A``
+      - ``2.1``
+
+    * - id
+      - ``N/A``
+      - ``autonomous-system--<object uuid>``
+
+    * - number
+      - ``data.metadata.asn``
+      - ``int(item["metadata"].get("asn").replace("AS", ""))``
```
Comment (zed-eiq) [inline extensions/greynoise/incoming_feeds/greynoise_noise_incoming_feed_v2/README.rst:294]: don't do this. this looks like business logic we have in the code. we should only document what this could look like in the end result, not the logic.

**Review by zed-eiq — CHANGES_REQUESTED** [review]
looks like in the STIX 2.1 column, you're neither referencing actual STIX 2.1 fields nor providing data formats -- but instead providing the business logic through which these values are obtained.

we should instead provide sample values, or human-friendly descriptions of expected values

File: extensions/greynoise/incoming_feeds/greynoise_noise_incoming_feed_v2/README.rst [hunk 10]
```diff
@@ -1,39 +1,624 @@
...
+      - ``2.1``
+
+    * - id
+      - ``N/A``
+      - ``identity--e5a3d1c3-e428-5f2a-b2bd-b2121e31d5c3``
+
+      
+    * - created
+      - Extension release date
+      - ``<Extension release date>`` (ISO8601 timestamp)
```
Comment (zed-eiq) [inline extensions/greynoise/incoming_feeds/greynoise_noise_incoming_feed_v2/README.rst:90]: there is an actual hardcoded value for this: https://github.com/eclecticiq/extensions-store/pull/89/files#diff-055ed765e9350a03c0257fb5be025693db918b727be8f0dbf0b0c5d4d47457aaR16

same for `modified` field

Comment (nathan-eiq) [reply extensions/greynoise/incoming_feeds/greynoise_noise_incoming_feed_v2/README.rst:90]: Isn't that value subject to change with the release of a new version?

File: extensions/greynoise/incoming_feeds/greynoise_noise_incoming_feed_v2/README.rst [hunk 11]
```diff
@@ -1,39 +1,613 @@
...
+GreyNoise Identity
+^^^^^^^^^^^^^^^^^^
+
+All STIX 2.1 objects produced by this extension will reference this Identity SDO as its producer (e.g. in their ``created_by_ref`` field).
+
+..  list-table::
+    :header-rows: 1
+    :stub-columns: 1
+
+    * - Field Name
```
Comment (zed-eiq) [inline extensions/greynoise/incoming_feeds/greynoise_noise_incoming_feed_v2/README.rst:76]: please apply review from https://github.com/eclecticiq/extensions-store/pull/119#pullrequestreview-1471666181

and to all the rest of the content here

File: extensions/greynoise/incoming_feeds/greynoise_noise_incoming_feed_v2/README.rst [hunk 12]
```diff
@@ -1,39 +1,613 @@
...
+    "API URL","api_url","string","true","The URL pointing to the API endpoint exposing the service that makes the data available for retrieval through the feed.","https://api.greynoise.io/"
+    "API Key","api_key","string","true","API Key","***"
+    "Classification","classification","array","true","Include indicators with selected classification","malicious"
+
+Data Mapping
+============
+
+Entities are transformed from GNQL json to `STIX 2.1 <https://docs.oasis-open.org/cti/stix/v2.1/stix-v2.1.html>`_ entities.
+
+STIX Domain Objects
```
Comment (zed-eiq) [inline extensions/greynoise/incoming_feeds/greynoise_noise_incoming_feed_v2/README.rst:64]: Please add the abbreviation

i.e. `STIX Domain Objects (SDO)`

also do this for the rest of your headings re: stix objects

**Review by zed-eiq — CHANGES_REQUESTED** [review]

File: extensions/greynoise/incoming_feeds/greynoise_noise_incoming_feed_v2/README.rst [hunk 13]
```diff
@@ -1,39 +1,613 @@
...
+
+Entities are transformed from GNQL json to `STIX 2.1 <https://docs.oasis-open.org/cti/stix/v2.1/stix-v2.1.html>`_ entities.
+
+STIX Domain Objects (SDO)
+-------------------------
+
+GreyNoise Identity
+^^^^^^^^^^^^^^^^^^
+
+All STIX 2.1 objects produced by this extension will reference this Identity SDO as its producer (e.g. in their ``created_by_ref`` field).
```
Comment (zed-eiq) [inline extensions/greynoise/incoming_feeds/greynoise_noise_incoming_feed_v2/README.rst:70]: as you've pointed out in the other PR! remove :pray:
